.github/workflows/tests.yaml

```diff
@@ -164,3 +161,12 @@ jobs:
             echo "configure script failure"
             exit 1
           fi
+      - name: Linux Multipass Provider Tests
+        run: |
+          sudo snap set charmcraft provider=multipass
+
+          pushd "$(mktemp -d)"
+          charmcraft -v pack --project-dir ../charm-smoke-test
+          test -f *.charm
+          charmcraft -v clean --project-dir ../charm-smoke-test
+          popd
```

Comment:
I know it is probably an irrelevant question, but is popd needed here? If it really is then it probably needs to be trapped.

Reply:
Trapped?  In this case I don't think the popd is required because its just one (last) test, but the pattern is appropriate I think.  Either way I removed this test because it has to run on macos to get nested virt.

Reply:
trap is like go's defer, you would do something like:

```
mkdir foo
trap "rm -rf foo" EXIT
pushd foo
exit 1
```

Reply:
AFAIK any command that fails will result in an abort of the remainder of the test

Reply:
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell